pkg/controller/machine.go

```diff
@@ -369,8 +368,9 @@ func (c *Controller) syncHandler(key string) error {
 }
 
 // deleteMachineAndProviderInstance makes sure that an instance has gone in a series of steps.
-func (c *Controller) deleteMachineAndProviderInstance(prov cloud.Provider, providerInstance instance.Instance, machine *machinev1alpha1.Machine) error {
+func (c *Controller) deleteMachineAndProviderInstance(prov cloud.Provider, machine *machinev1alpha1.Machine) error {
 	var err error
+	var providerInstance instance.Instance
```

Comment:
this cost me `nil pointer dereference` - golang shadowing :)

https://circleci.com/gh/kubermatic/machine-controller/2231?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link